Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs

```diff
@@ -0,0 +1,180 @@
+﻿// <copyright file="MSALPerUserMemoryTokenCache.cs" company="DNV GL Veracity">
+//  Licensed under the MIT License.   
+// </copyright>
+// <summary>
+//   An implementation of token cache for both Confidential and Public clients backed by MemoryCache
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Veracity.Authentication.OpenIDConnect.AspNet
+{
+    using System;
+    using System.Runtime.Caching;
+    using System.Security.Claims;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// An implementation of token cache for both Confidential and Public clients backed by MemoryCache.
+    /// MemoryCache is useful in Api scenarios where there is no HttpContext to cache data.
+    /// </summary>
+    /// <seealso cref="https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization"/>
+    public class MSALPerUserMemoryTokenCache
+    {
+        /// <summary>
+        /// The backing MemoryCache instance
+        /// </summary>
+        private readonly MemoryCache memoryCache = MemoryCache.Default;
+
+        /// <summary>
+        /// The duration till the tokens are kept in memory cache. In production, a higher value, upto 90 days is recommended.
```

Comment:
The lifetime of refresh token is one week in Veracity

Reply:
Fixed in a later check-in